priv/riak_kv.schema

```diff
@@ -968,44 +968,48 @@
      (_) -> false
   end}.
 
-%% @doc Register allowable origins (for CORS), specified as a list of ip
-%% ({host} or {host, port} tuple) under a scheme, i.e.
+%% @doc listener.$scheme.cors_allowable_origin.$origin is an IP address and TCP
+%% port that is allowed access to resources via Cross-Origin Resource Sharing
+%% (CORS).
 %%
-%% {riak_kv, [
-%%   {allowable_origin, {
-%%      http, [
-%%          {"127.0.0.1", 3000},
-%%          {"10.10.1.240"}
-%%      ],
-%%      https, [
-%%          {"127.0.0.1", 4430}
-%%      ]
-%%   }}
-%% ]}
-%%
-%% The resulting allowable origins are "http://127.0.0.1:300",
-%% "http://10.10.1.240:*", "https://127.0.0.1:4430".
-%%
-%% The special case of "Allow All" is achieved by specifying the scheme as '*'
-%% (wildcard), i.e.
-%%
-%% {riak_kv, [
-%%   {allowable_origin, { '*' } }
-%% ]}
-%%
-%% If port is not specified, all ports for the host are to be considered allowed.
-%% This is equivalent to specifying the port using the wildcard.
-{mapping, "allowable_origin.$name", "riak_kv.allowable_origin.http", [
-    {datatype, ip},
-    {default, {}},
-    {level, advanced}
+%% Allowable schemes include http and https, and all where the wildcard (all)
+%% is used to allow all origins to share resources with Riak.
+{mapping, "listener.$scheme.cors_allowable_origin.$origin", "riak_api.cors_allowable_origins",
+[
+ {datatype, [ip, flag]}
```

Comment:
@angrycub 
wrt splitting out `all`since it is the only flag datatype does have a indisputable benefit, so I will make that change.

I disagree wrt splitting out `http` and `https`, this design results in duplicated code as in listener.http.$name. IME the runtime cost incurred at startup to find and throw if any invalid schemes are present is cheap, especially when compared to the need to hotfix when someone fails to keep the code in sync between the repeated http and https config code AND config code does not lend well to testing, so compounded. If you feel that I am wrong, let me know and I'll make the change.